### ISSUE TITLE
Not working for v6.4

### ISSUE THREAD

**Issue by bharatm89** [snuids/TrafficLightVisKibana6#1]
Looks like this plug-in is not working for kibana v6.4 .  can you confirm ?

**Comment by pclohessy** [comment]
It doesn't appear to work with Kibana 6.5 either. 

The graphics are blank.

![trafficlightviskibana6 - icon](https://user-images.githubusercontent.com/46590534/51019703-ee6f1800-15b6-11e9-8356-c537b26bdee9.png)
![trafficlightviskibana6 - new visualization](https://user-images.githubusercontent.com/46590534/51019704-f0d17200-15b6-11e9-9d05-7fa0d6c4c0bf.png)

**Comment by gaby** [comment]
I'm having the same issue with Kibana 6.4.x and 6.5.x

**Comment by snuids** [comment]
This is strange. I am using it with Kibana 6.4.1. Do you have some errors in your browser console ?

**Comment by bharatm89** [comment]
I was using v6.4.2 and i dont see trafficLight  under visualizations though its copied under plugin folder of kibana

**Comment by gaby** [comment]
No errors in the console or notifications in Kibana. I just tested in 6.5.4 and it doesnt work. The visualization is empty like the one @pclohessy posted. 

The way I installed this was by copying the git repo to the Kibana plugins folder and restarting Kibana.

**Comment by bharatm89** [comment]
I tried in v6.4.1 and i see same issue . Blank screen when i click on visualization and if i remove plugin from kibana plugin folder- kibana works fine.

![image](https://user-images.githubusercontent.com/46569350/51065155-018ae380-15c9-11e9-8297-2857ccedd0a1.png)

I see below error in browser console.
![image](https://user-images.githubusercontent.com/46569350/51066209-7b719b80-15ce-11e9-9199-7ade829c3e2e.png)

I see similar issue  here - 
https://github.com/elastic/kibana/pull/17565

Any idea if this can be fixed for trafficLights?

**Comment by snuids** [comment]
@bharatm89
Made a branch 6.5 that seems to work for me (6.5.4). Can you check please ?
Don't know if it works with versions between 6.4.1 and 6.5.3.

Cheers,

**Comment by bharatm89** [comment]
@snuids I am getting same error for 6.5.4 as well .
I downloaded it from 6.5 branch .I am trying it on my windows local desktop.

![image](https://user-images.githubusercontent.com/46569350/51081431-e5647080-16b4-11e9-846d-c574e399ef78.png)


![image](https://user-images.githubusercontent.com/46569350/51081422-c5cd4800-16b4-11e9-90fb-8e0e17fddea8.png)

**Comment by snuids** [comment]
Hi,

I remade a full install from scratch on my mac.

* Downloaded the Kibana 6.5.4 package from Elastic.
* Downloaded the branch 6.5 of the plugin.
* Copied the plugin in the Plugin directory and started Kibana.

After a few minutes everything was Ok.
Note that the icon of the plugin should look as shown on the screenshot.

Try perhaps to reload Kibana in a chrome private window to ensure that your browser cache is empty?

Cheers,

<img width="1680" alt="screenshot 2019-01-13 at 11 15 38" src="https://user-images.githubusercontent.com/13175107/51084106-1086a880-1725-11e9-979d-b7ceb2a97ab5.png">

**Comment by gaby** [comment]
The plugin seems to be working fine with Kibana 6.5.4 Docker Image. The only problem I have right now is that no matter which aggregation, I only get 1 traffic light that says "All". I'm testing with Elstic Hearbeat.

**Comment by snuids** [comment]
@gabrielcalderon

In fact I am writing "All" if you don't split the graph by buckets.
In fact I never used the plugin without splitting. For me the most interesting thing in this plugin is the option to split traffic lights on multiple lines and columns.

Do you split your serie by term ?

I should be smarter than that. I will try to fix that this week. Thanks for the return.

Cheers,

**Comment by gaby** [comment]
I try splitting the series, but since I only had 1 result it said "All" instead of the value of the splitting field. This may be a bug? 

The other thing I noticed, when the metric value equals the max, the traffic light is red instead of green.

**Comment by snuids** [comment]
I will try to fix this, this week. I have a rough idea of where the problem is.

Cheers

**Comment by gaby** [comment]
@snuids I found a fix for the problem, i'm submitting a PR in branch 6.5.

**Comment by bharatm89** [comment]
@snuids  Thanks for your help , it worked for v6.5.4
Here is what i am trying to do .
I have few saved queries which search for specific container-name and AUDIT logs . 
Example: (kubernetes.container_name:"xyz" OR kubernetes.container_name:"abc" OR kubernetes.container_name:"pqr" ) AND json_logType:"AUDIT"

I want to create one trafficSignal light and if i get more 400/500 errors ( based on threshold) it should turn RED/Yellow and otherwise it should be Green.

I tried to add Terms/filters on status-code but its adding multiple trafficLights , Is there way i can use Json input in advance options and see just one traffic light ?

Thanks for your help again in advance.

**Comment by gaby** [comment]
@bharatm89 I think you can do that using a scripted field in Kibana. Management -> Index Pattern -> Scripted field. Scripted fields use painless scripting so you can do like:

if ( doc['docker.container.name'].value == "XYZ" ) { return doc['status.code'].value; }

**Comment by snuids** [comment]
@gabrielcalderon 

Merged your PR.

Thanks

**Comment by gaby** [comment]
No problem!

**Comment by bharatm89** [comment]
Anyone able to make it work for 6.4.0 ?

**Comment by gaby** [comment]
@bharatm89 Did you try using the latest from the 6.5 branch?

**Comment by bharatm89** [comment]
yes i tried using the latest one from 6.5 branch and master branch both . 
I am getting below error in kibana 6.4.0 when i try to start server

![image](https://user-images.githubusercontent.com/46569350/51152866-5fbaff00-1833-11e9-8c0c-2bb768bfdc19.png)

If i remove plugins - kibana just works fine.

**Comment by snuids** [comment]
@ bharatm89

I was using the master branch for Kibana 6.4.1 and the branch 6.5 for Kibana 6.5.4. I checked them using docker containers on linux and native installations on my mac without problem. I don't have any windows computer to check it on Windows. From what I understand, for you:

* branch 6.5 works with Kibana 6.5 on your windows
* branch 6.5 fails with kibana 6.4 on your windows
* master branch fails with kibana 6.4 on your windows

Am I right ?
Can you specify exactly the versions of your kibanas ?

Cheers,

**Comment by bharatm89** [comment]
Yes thats right- all 3 points.
I am looking for this plugin for kibana version 6.4.0.

I am getting below error with master/6.5 plugin and kibana is failing to start.

![image](https://user-images.githubusercontent.com/46569350/51219931-b0486000-18f8-11e9-889d-60b1a67d3397.png)

@snuids  Can you confirm if it ever worked on Kibana 6.4.0 for you ?



Thanks,
Bharat

**Comment by gaby** [comment]
@bharatm89 See this: https://discuss.elastic.co/t/error-cannot-delete-files-folders-outside-the-current-working-directory/148702

Try running kibana with: 

`bin/kibana --force`

**Comment by bharatm89** [comment]
@gabrielcalderon : yes i am getting same old error .
Not sure if it worked for anyone on  Kibana 6.4.0 ? 

![image](https://user-images.githubusercontent.com/46569350/51225574-d4b03680-1910-11e9-96c9-52188cb49f71.png)

**Comment by snuids** [comment]
@bharatm89 

You are right there is something wrong with kibana 6.4.0. I get the same error as you with this version (Cannot delete). With version 6.4.1, it starts but the graphics are blank. I ,need more time to investigate. The quickest fix for you is to probably upgrade to 6.5.

Cheers,

**Comment by bharatm89** [comment]
@snuids 

For some reason i can't upgrade Kibana to v6.5 and i will have to stick with v6.4.0 , I tried to fix it but its not working :(  , please let me know if you can fix it for v6.4.0 .

**Comment by snuids** [comment]
@bharatm89

I don't think it is possible for 6.4.0. It looks like there is a problem in this particular version that crashes at startup as soon as a plugin is installed. I did not check with other plugins, but I suppose that it will be the same problem.

On 6.4.1, it starts but it does not display anything. I don't know if it can be fixed. 

To be honest that's really difficult to write plugins as Elastic Search changes the APIs every minor version.

Cheers,
